libavformat/mpegtsenc.c

```diff
@@ -1551,9 +1551,24 @@ static void mpegts_write_pes(AVFormatContext *s, AVStream *st,
     ts_st->prev_payload_key = key;
 }
 
+static int is_h264_start_code_3bytes(const uint8_t* p) 
+{
+    if (p[0] == 0x0 && p[1] == 0x00 && p[2] == 0x01) return 1;
+    return 0;
+}
+
+static int can_find_h264_start_code(const uint8_t* p, size_t size) 
```

Comment:
Is this covering the 4 byte case as well?

Reply:
yeah. With the 3 byte code, we will need an extra execution
For instance, with "0, 0, 0, 1", the first execution will get 0, 0, 0, so false. We advance the pointer and we get 0, 0 ,1, so true